RevenueCatUI/Views/RemoteImage.swift

```diff
@@ -17,44 +17,76 @@ import SwiftUI
 struct RemoteImage: View {
 
     let url: URL
+    let lowResUrl: URL?
     let aspectRatio: CGFloat?
     let maxWidth: CGFloat?
 
+    var fetchLowRes: Bool {
+        lowResUrl != nil
+    }
+
+    @StateObject
+    private var highResLoader: ImageLoader = .init()
+
     @StateObject
-    private var loader: ImageLoader = .init()
+    private var lowResLoader: ImageLoader = .init()
 
-    init(url: URL, aspectRatio: CGFloat? = nil, maxWidth: CGFloat? = nil) {
+    init(url: URL, lowResUrl: URL? = nil, aspectRatio: CGFloat? = nil, maxWidth: CGFloat? = nil) {
         self.url = url
         self.aspectRatio = aspectRatio
         self.maxWidth = maxWidth
+        self.lowResUrl = lowResUrl
     }
 
     var body: some View {
         Group {
-            switch self.loader.result {
-            case .none:
-                self.emptyView(error: nil)
-
-            case let .success(image):
-                if let aspectRatio {
-                    image
-                        .fitToAspect(aspectRatio, contentMode: .fill)
-                        .frame(maxWidth: self.maxWidth)
-                        .accessibilityHidden(true)
-
+            if case let .success(image) = highResLoader.result {
+                displayImage(image)
+            } else if case let .success(image) = lowResLoader.result {
+                displayImage(image)
+            } else if case let .failure(highResError) = highResLoader.result {
+                if !fetchLowRes {
+                    emptyView(error: highResError)
+                } else if case .failure = lowResLoader.result {
+                    emptyView(error: highResError)
                 } else {
-                    image
-                        .resizable()
-                        .accessibilityHidden(true)
+                    emptyView(error: nil)
                 }
-
-            case let .failure(error):
-                self.emptyView(error: error)
+            } else {
+                emptyView(error: nil)
             }
         }
         .transition(Self.transition)
         .task(id: self.url) { // This cancels the previous task when the URL changes.
```

Comment:
This should probably be 2 separate tasks. This is unnecessarily cancelling loading `url` if the other UEL changes.

Reply:
Kinda feel like that might be more of a feature than a bug? Like, if one of the image URLs changes, it's more than likely than both would... And if they didn't, it might not be a bad idea to still reload both just in case the URL remains the same but not the image. 

PS: Hi Nacho 👋

Reply:
I kind of like the idea...

```swift
        .task(id: self.url) {
            await loadHighResImage()
        }
        .task(id: self.lowResUrl) {
            await loadLowResImage()
        }
```
calling
```swift
    private func loadHighResImage() async {
        await highResLoader.load(url: url)
    }

    private func loadLowResImage() async {
        guard let lowRes = lowResUrl else { return }
        await lowResLoader.load(url: lowRes)
    }
```
which seems more elegant than the logic currently inside `loadImages()`:

```swift
    private func loadImages() async {
        if fetchLowRes, let lowResLoc = lowResUrl {
            async let lowResLoad: Void = lowResLoader.load(url: lowResLoc)
            async let highResLoad: Void = highResLoader.load(url: url)
            _ = await (lowResLoad, highResLoad)

        } else {
            await highResLoader.load(url: url)
        }
    }
```

BUT keeping `.url` as the sole identifier I feel makes some sense too.

Reply:
Yes that implementation makes sense 👍🏻